dvc/fs/dvc.py

```diff
@@ -137,12 +138,21 @@ def _getcwd():
         if hasattr(repo, "dvc_dir"):
             self._datafss[key] = DataFileSystem(index=repo.index.data["repo"])
 
-    def _get_key(self, path):
+    def _get_key(self, path) -> Key:
```

Comment:
Would lru help here?

Reply:
The goal is to perform this operation only once per path, so I guess not, but I'll keep it in mind.